src/NavigatorContainer.tsx

```diff
@@ -31,17 +32,29 @@ export class NavigatorContainer extends React.Component<Props> {
 
   componentWillMount() {
     BackHandler.addEventListener('hardwareBackPress', this.navigateBack)
+    if (Platform.OS === 'android') {
+      Linking.getInitialURL().then((url: string) => {
+        this.props.handleDeepLink(url)
+      })
+    } else {
+      Linking.addEventListener('url', this.handleOpenURL)
+    }
   }
 
   componentWillUnmount() {
     BackHandler.removeEventListener('hardwareBackPress', this.navigateBack)
+    Linking.removeEventListener('url', this.handleOpenURL)
   }
 
   private navigateBack = () => {
     this.props.goBack()
     return true
   }
 
+  private handleOpenURL = (event: any) => {
```

Comment:
Nitpick, can we inline this on line 40? Or perhaps use it on line 37.

Reply:
We need this additional function as in `addListener` function, the callback gets an event, and we need to extract url from an event. So we can't inline that.